src/ZiraLink.Api.Application/Services/AppProjectService.cs

```diff
@@ -0,0 +1,125 @@
+                CustomerId = customer.Id,
+                Title = title,
+                AppProjectType = appProjectType,
+                InternalPort = internalPort,
+                DateCreated = DateTime.UtcNow,
+                DateUpdated = DateTime.UtcNow,
+                State = state
+            };
+
+            await _dbContext.AppProjects.AddAsync(appProject, cancellationToken);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+            _bus.Publish("APP_PROJECT_CREATED");
+
+            return appProject.Id;
+        }
+
+        public async Task DeleteAsync(long customerId, long id, CancellationToken cancellationToken)
+        {
+            var appProjects = await _dbContext.AppProjects.AsNoTracking().Where(x => x.Id == id && x.CustomerId == customerId).SingleOrDefaultAsync(cancellationToken);
+            if (appProjects == null)
+                throw new NotFoundException(nameof(AppProject));
+            _dbContext.ChangeTracker.Clear();
+            _dbContext.AppProjects.Remove(appProjects);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+            _bus.Publish("APP_PROJECT_DELETED");
+        }
+
+        public async Task PatchAsync(long id, long customerId, string title, Guid? appProjectViewId, AppProjectType appProjectType, int internalPort, RowState state, CancellationToken cancellationToken)
+        {
+            if (appProjectType == AppProjectType.UsePort && (!appProjectViewId.HasValue || appProjectViewId == Guid.Empty))
+                throw new ArgumentNullException(nameof(appProjectViewId));
+
+            var customer = await _dbContext.Customers.AsNoTracking().SingleOrDefaultAsync(x => x.Id == customerId, cancellationToken);
+            if (customer == null)
+                throw new NotFoundException(nameof(Customer), new List<KeyValuePair<string, object>>() { new KeyValuePair<string, object>(nameof(Customer.ExternalId), customerId) });
+
+            var appProject = await _dbContext.AppProjects.SingleOrDefaultAsync(x => x.Id == id && x.CustomerId == customerId, cancellationToken);
+            if (appProject == null)
+                throw new NotFoundException(nameof(AppProject), new List<KeyValuePair<string, object>> { new KeyValuePair<string, object>(nameof(AppProject.Id), id) });
+             
+            //Guid appProjectViewIdGuid = string.IsNullOrWhiteSpace(appProjectViewId) ? Guid.Empty : Guid.Parse(appProjectViewId);
+
+            if (appProjectType == AppProjectType.UsePort && !_dbContext.AppProjects.Any(x => x.ViewId == appProjectViewId))
+                throw new NotFoundException(nameof(AppProject), new List<KeyValuePair<string, object>> { new KeyValuePair<string, object>(nameof(AppProject.AppProjectViewId), appProjectViewId.Value) });
+
+            if (!string.IsNullOrWhiteSpace(title))
+                appProject.Title = title;
+            if (internalPort < 1 || internalPort > 65535)
```

Comment:
The condition here should be `if (port >= 1 && port <=65535) then ...`

Reply:
Done.